packages/design-system-react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,119 @@
+import type { ChangeEvent, KeyboardEvent } from 'react';
+import React, { forwardRef, useImperativeHandle } from 'react';
+
+import { twMerge } from '../../utils/tw-merge';
+import { Icon, IconName, IconColor, IconSize } from '../Icon';
+import { Text } from '../Text';
+
+import type { CheckboxProps } from './Checkbox.types';
+
+export const Checkbox = forwardRef<{ toggle: () => void }, CheckboxProps>(
+  (
+    {
+      id,
```

Comment:
Made `id` a required prop after noticing this issue in the extension component. Chrome has come out with accessibility tooling highly recommended turning it on from experiemental settings

<img width="1512" alt="Screenshot 2025-05-23 at 11 42 16 AM" src="https://github.com/user-attachments/assets/50d780b0-527b-4389-b0c7-f704e444ab67" />